src/memtable.rs

```diff
@@ -31,32 +74,34 @@ impl Drop for MemTableView {
     }
 }
 
-pub struct MemTable {
-    mutable: Skiplist<Flsc>,
-    immutable: VecDeque<Skiplist<Flsc>>,
+pub struct MemTables {
+    mutable: MemTable,
+    immutable: VecDeque<MemTable>,
 }
 
-impl MemTable {
-    pub fn with_capacity(table_size: u32, max_count: usize) -> MemTable {
-        let c = Flsc::new(8);
-        MemTable {
-            mutable: Skiplist::with_capacity(c, table_size),
-            immutable: VecDeque::with_capacity(max_count - 1),
-        }
+impl MemTables {
+    pub(crate) fn new(mutable: MemTable, immutable: VecDeque<MemTable>) -> Self {
```

Comment:
I prefer the previous `with_capacity` instead of new with `mutable` and `immutable` from outside

Reply:
The "open memtable" logic is implemented in `db.rs`, the database engine. Therefore, it's necessary to provide an interface to construct `Memtables` from a single `Memtable`.